### PR TITLE
Add `intersects?` to `Relation`

### PR DESCRIPTION
**Pull request by john-h-k**
Ruby 3.1 added `intersects?` which is equivalent to `(a & b).any?`. Rubocop added a corresponding cop, `Style/ArrayIntersect`, which transforms the old style to use `intersects?`. Unfortunately as `intersects?` is not delegated on `CollectionProxy`, this leads to false positives that need to be disabled for no good reason other than the fact the method isn't delegated.

This PR add delegation of `intersects?` to `Relation` which fixes this.

`intersection` and `union` are identical to `&` and `|` respectively (which are already delegated), so I thought for symmetry it was worth adding these, although I am not against removing them from this PR if it makes it more likely to be accepted.

### Motivation / Background

This Pull Request has been created because [REPLACE ME]

### Detail

This Pull Request changes [REPLACE ME]

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by matthewd** [comment]
> I thought for symmetry it was worth adding [`intersection` and `union`], although I am not against removing them from this PR if it makes it more likely to be accepted

Yes please. Despite them being aliases for the already-defined Array operations, as you note, I think I'd prefer to keep those names free just in case we ever implement full-scale SQL set operations on Relation. `intersect?` seems fine, because its return value remains the same regardless of how we generate the answer -- but if we define `Relation#union(Relation) -> Array` now, it would be extremely difficult to later change it to returning a Relation.

**Comment by john-h-k** [comment]
Have done that now @matthewd

**Review by skipkayhil — COMMENTED** [review]
Can you [squash your commits](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#squashing-commits) as well?

**Comment by john-h-k** [comment]
Squashed

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+*   Add support for `Array#intersect?` to `ActiveRecord::Relation`.
```
Comment (brandondrew) [inline activerecord/CHANGELOG.md:1]: This file uses `intersect?` (without an 'S'), but the other files use `intersects?` (with an 'S').  Perhaps that can be corrected, to help avoid bugs when people copy the method name from the change log?

Comment (byroot) [reply activerecord/CHANGELOG.md:1]: Good catch. Fixed in https://github.com/rails/rails/commit/e446ef82c8174a4bd6c98f670912bb395b3091d6, thank you.

Comment (guilleiguaran) [reply activerecord/CHANGELOG.md:1]: The CHANGELOG is correct, I fixed the code in https://github.com/rails/rails/commit/119b4b2af1d848362a1a9c1cd8a2bea46700b154

**Pull request merged**